api-fput-object.go

```diff
@@ -139,49 +141,100 @@ func (c Client) FPutObject(bucketName, objectName, filePath, contentType string)
 		return 0, err
 	}
 
-	// gather next part number to be uploaded, total uploaded size and list of all parts uploaded.
-	partNumber, totalUploadedSize, completeMultipartUpload, err := c.getMultipartStat(bucketName, objectName, uploadID)
+	// total data read and written to server. should be equal to 'size' at the end of the call.
+	var totalUploadedSize int64
+
+	// Complete multipart upload.
+	var completeMultipartUpload completeMultipartUpload
+
+	// Fetch previously upload parts and save the total size.
+	partsInfo, err := c.listObjectParts(bucketName, objectName, uploadID)
 	if err != nil {
 		return 0, err
 	}
+	// Save uploaded parts in a map.
+	var uploadedPartsMap = make(map[int]completePart)
+	// Previous maximum part size
+	var prevMaxPartSize int64
+	// Loop through all parts to save them in a local map.
+	for _, partInfo := range partsInfo {
+		var completedPart completePart
```

Comment:
requires code comment. Particularly explain that how part size chosen based on max part size.